lib/ansible/galaxy/dependency_resolution/dataclasses.py

```diff
@@ -43,36 +43,47 @@
 display = Display()
 
 
-def get_validated_source_info(b_source_info_path, namespace, name, version):
+def get_validated_source_info(
+        b_source_info_path: bytes,
+        namespace: str,
+        name: str,
+        version: str,
+) -> dict[str, object] | None:
     source_info_path = to_text(b_source_info_path, errors='surrogate_or_strict')
 
     if not os.path.isfile(b_source_info_path):
         return None
 
     try:
         with open(b_source_info_path, mode='rb') as fd:
-            metadata = safe_load(fd)
+            loosely_typed_metadata = safe_load(fd)
     except OSError as e:
         display.warning(
             f"Error getting collection source information at '{source_info_path}': {to_text(e, errors='surrogate_or_strict')}"
         )
         return None
 
-    if not isinstance(metadata, MutableMapping):
+    if not isinstance(loosely_typed_metadata, MutableMapping):
```

Comment:
Update this `isinstance` check to use `dict` instead of `MutableMapping`, since that's what it should be. That will eliminate the need for the type cast below.

Reply:
IIRC it didn't work. But I'll double-check.